SpatialGDK/Source/SpatialGDK/Private/Interop/GlobalStateManager.cpp

```diff
@@ -589,24 +623,26 @@ void UGlobalStateManager::QueryGSM(bool bRetryUntilAcceptingPlayers)
 		}
 		else
 		{
-			bool bNewAcceptingPlayers = GetAcceptingPlayersFromQueryResponse(Op);
+			bool bNewAcceptingPlayers = false;
+			int32 QuerySessionId = 0;
+			bool bQueryResponseSuccess = GetAcceptingPlayersAndSessionIdFromQueryResponse(Op, bNewAcceptingPlayers, QuerySessionId);
+			bool bChecksAreGood = bQueryResponseSuccess && bNewAcceptingPlayers == AcceptingPlayersToCheck && QuerySessionId == SessionIdToCheck;
```

Comment:
I'm finding this a bit hard to parse xD
Let's break it up. We should probably check `bQueryResponseSuccess` separately to the values of `bNewAcceptingPlayers` and `QuerySessionId` right? As the `bQueryResponseSuccess` is based on whether the data is malformed or not and thus should have a warning or error if it goes wrong (we should still retry in this case but better logging for debugging).
Also a better name for `bChecksAreGood` would be good :D

Reply:
In the case that we get `bQueryResponseSuccess=true` we should:
```suggestion
				UE_LOG(LogGlobalStateManager, Log, TEXT("GlobalStateManager query result: AcceptingPlayers=%s, SessionID=%s"), bNewAcceptingPlayers, QuerySessionId);
```